drivers/usb/usbstor/scsi.c

```diff
@@ -525,6 +560,46 @@ USBSTOR_SendCBWRequest(
         Context);
 }
 
+static
+NTSTATUS
+USBSTOR_IssueRequestSense(
+    IN PFDO_DEVICE_EXTENSION FDODeviceExtension,
+    IN PIRP Irp,
+    IN PIRP_CONTEXT Context)
+{
+    PIO_STACK_LOCATION IoStack;
+    PSCSI_REQUEST_BLOCK CurrentSrb;
+    PSCSI_REQUEST_BLOCK SenseSrb;
+
+    DPRINT("USBSTOR_IssueRequestSense: \n");
+
+    CurrentSrb = FDODeviceExtension->ActiveSrb;
+    SenseSrb = &Context->SenseSrb;
+    IoStack = IoGetCurrentIrpStackLocation(Irp);
+    IoStack->Parameters.Scsi.Srb = SenseSrb;
+
+    RtlZeroMemory(SenseSrb, sizeof(Context->SenseSrb));
+
+    SenseSrb->Function = SRB_FUNCTION_EXECUTE_SCSI;
+    SenseSrb->Length = sizeof(Context->SenseSrb);
+    SenseSrb->CdbLength = 12;
+
+    SenseSrb->SrbFlags = SRB_FLAGS_DATA_IN |
+                         SRB_FLAGS_NO_QUEUE_FREEZE |
+                         SRB_FLAGS_DISABLE_AUTOSENSE;
+
+    ASSERT(CurrentSrb->SenseInfoBufferLength);
+    ASSERT(CurrentSrb->SenseInfoBuffer);
+
+    SenseSrb->DataTransferLength = CurrentSrb->SenseInfoBufferLength;
+    SenseSrb->DataBuffer = CurrentSrb->SenseInfoBuffer;
+
+    SenseSrb->Cdb[0] = SCSIOP_REQUEST_SENSE;
```

Comment:
This should ideally use the CDB structure (see other code that does `pCDB = (PCDB)Srb->Cdb` etc